tapis-project/authenticator
# enhance: gh-40 new design for IDP form

## Overview

Redesign the IDP form.

## Related

- closes #40
- requires https://github.com/TACC/Core-Styles/pull/230
- requires https://github.com/TACC/Core-Styles/pull/227
- fixed by #57

## Changes

- **updated** version of core-styles
- **added** icons (SVG files)
- **added** custom styles for buttons
- **changed** text
- **fixed** error markup
- **changed** radio buttons to be submit buttons
- **removed** primary submit button

## Testing

1. Load form.
2. Verify form matches expected UI.
3. Verify each buttons perform the same as did the old respective radio button choice plus old single submit button.

> **Note**
> I only tested №2 _in situ_ using expected markup + locally-served assets.
> <details><summary>HTML</summary>
>
> ```html
> <main class="s-form-page">
>   <style>
>   .button-list {
>     /* To lay out buttons in a flexible centered column */
>     display: flex;
>     align-items: center;
>     flex-direction: column;
>     gap: 0.5em;
>   }
>   .button-list button {
>     min-width: 40rem; /* to limit button width but allow wider */
>
>     /* To position button content */
>     text-align: start;
>     padding-left: 1em;
>   }
>   .button-list button img {
>     width: 8rem;
>     height: 4rem;
>     margin-right: 0.5em;
>   }
>   </style>
>   <form action="" method="post" class="s-form s-form--login">
>     <h1>
>       <figure>
>         <img src="authorize/dev.jpeg" alt="Dev">
>         <img src="authorize/tacc-formal.svg" alt="TACC">
>       </figure>
>       <span>Log in to Tapis</span>
>     </h1>
>     <p>Select an account to continue</p>
>     <div>
>       <menu class="button-list">
>         <li>
>           <button name="idp_id" value="prod_kc" type="submit">
>             <img src="http://localhost:51812/prod_kc.svg" alt="TACC Keycloak" />
>             Continue with TACC Keycloak</button>
>         </li>
>         <li>
>           <button name="idp_id" value="dev_kc" type="submit">
>             <img src="http://localhost:51812/dev_kc.svg" alt="Dev keycloak" />
>             Continue with Dev keycloak</button>
>         </li>
>         <li>
>           <button name="idp_id" value="dev_ldap" type="submit">
>             <img src="http://localhost:51812/dev_ldap.svg" alt="Test Accounts" />
>             Continue with Test Accounts</button>
>         </li>
>         <li>
>           <button name="idp_id" value="github" type="submit">
>             <img src="http://localhost:51812/github.svg" alt="GitHub Accounts" />
>             Continue with GitHub Accounts</button>
>         </li>
>         <li>
>           <button name="idp_id" value="globus" type="submit">
>             <img src="http://localhost:51812/globus.svg" alt="Globus Auth" />
>             Continue with Globus Auth</button>
>         </li>
>       </menu>
>     </div>
>     <nav>
>       <p>Having trouble?</p>
>       <a href="https://tacc.utexas.edu/about/help/" rel="noreferrer" target="_blank"> Account Help </a>
>     </nav>
>   </form>
>   <footer>
>     <a href="https://tacc.utexas.edu/about/security-and-compliance/"> Security </a>
>     <a href="https://tacc.utexas.edu/use-tacc/user-policies/"> Policies </a>
>   </footer>
> </main>
> ```
>
> </details>

## UI

| expected UI |
| - |
| <img width="992" alt="with icons" src="https://github.com/tapis-project/authenticator/assets/62723358/ed73694a-54da-43fc-bf02-ccbb564a476b"> |

<details><summary>Archive</summary>

| as of 7ebf628 (no icons yet) |
| - |
| <img width="1200" alt="idp redesign - sans icons" src="https://github.com/tapis-project/authenticator/assets/62723358/318a8deb-9176-47d0-b0d6-f454a88e6790"> |

</details>
